File: fact.js
#!/usr/bin/env node
"use strict";

/*
    @@@ get_version: node --version | sed s/v//
    @@@ is_fast: 1
*/

var times = 50000000;
var fact_16 = 20922789888000;

function fact(n) {
    if (n == 1)
        return 1;
    return n * fact(n - 1);
};

var ok = true;
for (var i = 0; i < times; i++) {
    ok = ok && fact(16) == fact_16;
};

console.log(`node.js finish ${times} - ${ok ? 'ok' : 'fail'}`);

File: fact.pl
#!/usr/bin/perl

# @@@ get_version: perl -v | perl -ne 'print $1 if /(\d+(\.\d+){2})/'

use 5.10.0;

use strict;
use warnings;

my $times = 1_000_000;
my $fact_16 = 20922789888000;

# ------------------------------------------------------------------------------
sub fact {
    my $n = shift;
    return 1 if $n == 1;
    return $n * fact($n - 1);
}

my $ok = 1;
$ok &&= fact(16) == $fact_16 for 1 .. $times;
say "perl finish $times - " . ($ok ? 'ok' : 'fail');
